docs/_tooling/extensions/header_service/test/test_header_service_integration.py

```diff
@@ -0,0 +1,238 @@
+        prefix = "RH__"),
+]
+needs_id_regex = ".*"
+needs_extra_options = [
+    "security",
+    "safety",
+    "rationale",
+    "reqtype",
+    "codelink",
+    "testlink",
+    "reqcovered",
+    "testcovered",
+    "hash",
+    "author",
+    "reviewers",
+    "approvers",
+]
+header_service_use_github_data = {use_github_data}
+"""
+
+    return wrapper
+
+
+@pytest.fixture(scope="session")
+def basic_needs():
+    return """
+TESTING HEADER SERVICE
+======================
+
+.. needservice:: header-service
+
+"""
+
+
+@pytest.fixture(scope="session")
+def template_needs():
+    return """
+.. list-table::
+   :widths: 50 50
+   :header-rows: 0
+
+   * - **DOCUMENT IDENTIFICATION**
+     -
+   * - Document Type
+     - Checklist
+   * - Document ID
+     - DPX-CONTR-REVIEW-CHECKLIST
+   * - Project Name
+     - Dependix
+   * - ASIL
+     - B
+   * - Security Classification
+     - CONFIDENTIAL
+   * - Author
+     - {{ author }}
+   * - Reviewer
+     -
+{%- for r in reviewers.split(", ") %}
+       | {{ r -}}
+{% endfor %}
+   * - Approver
+     -
+{%- for a in approvers.split(", ") %}
+       | {{ a -}}
+{% endfor %}
+   * - Version
+     - {{ hash }}
+   * - Status
+     - RELEASED
+"""
+
+
+# @patch("header_service.header_service.generate_hash")
+# @patch("header_service.header_service._extract_github_data")
+# @patch("sphinx.application.Sphinx")
+# def test_request_from_directive_github_data(
+#     mock_app, mock_extract_github_data, mock_generate_hash
+# ):
+
+# @patch("header_service.header_service.generate_hash")
+# @patch("header_service.header_service._extract_merge_commit_data")
+# @patch("sphinx.application.Sphinx")
+# def test_request_from_directive_commit_data(
+#     mock_app, mock_extract_merge_commit_data, mock_generate_hash
+# ):
+
+
+@patch("header_service.header_service.generate_hash")
+@patch("header_service.header_service._extract_github_data")
+def test_header_service_integration_github_data(
+    mock_extract_github_data,
+    mock_generate_hash,
+    sphinx_app_setup,
+    basic_conf,
+    basic_needs,
+    template_needs,
+):
+    github_data = {
+        "author": "John Doe",
+        "approvers": "aprover_1",
+        "reviewers": "reviewer_1",
+        "hash": "abcdef",
+    }
+    mock_generate_hash.return_value = "abcdef"
+    mock_extract_github_data.return_value = github_data
+    mock_basic_conf = basic_conf()
+    app = sphinx_app_setup(
+        mock_basic_conf,
+        basic_needs,
+        template_needs,
+    )
+    try:
+        # app.env.config.header_service_use_github_data = True
+        app.build()
+        html_content = (app.outdir / "index.html").read_text()
+        mock_extract_github_data.assert_called_once()
+        assert "<td><p>John Doe</p></td>" in html_content
+        assert '<div class="line">aprover_1</div>' in html_content
+        assert '<div class="line">reviewer_1</div>' in html_content
+        assert "<td><p>abcdef</p></td>" in html_content
```

Comment:
With these tests in place, tests like `test_register` can be removed as they are redundant. This test verifies results, while `test_register` verifies internal details which can change with every sphinx release etc.